src/main/resources/META-INF/resources/primefaces/datepicker/0-datepicker.js

```diff
@@ -17,6 +17,12 @@
     }
 }(function ($) {
 
+    function calculateWeekNumber(d) {
+        var yearStart = new Date(Date.UTC(d.getUTCFullYear(),0,1));
+        var weekNo = Math.ceil(( ( (d - yearStart) / 86400000) + 1)/7);
+        return weekNo;
+    }
```

Comment:
Why not put this method right below the 3 methods using it?

Reply:
I don't think that it would be a good idea to invoke the calculation method with the `this` scope. So I can't add the common method to the widget because then I habe to invoke it with `this`.

AT least that is my understanding, but I have to test it.

Reply:
yeah and if you consider it a private method of the widget you can name it with an underscore so `_calculateWeekNumber` that is how other widgets do methods like these as internal helper methods..

Reply:
Also, why UTC? could you please remove UTC format?

https://github.com/primefaces/primereact/blob/master/src/components/calendar/Calendar.js#L1586
```
getWeekNumber(date) {
        let checkDate = new Date(date.getTime());
        checkDate.setDate(checkDate.getDate() + 4 - ( checkDate.getDay() || 7 ));
        let time = checkDate.getTime();
        checkDate.setMonth( 0 );
        checkDate.setDate( 1 );
        return Math.floor( Math.round((time - checkDate.getTime()) / 86400000 ) / 7 ) + 1;
    }
```

Reply:
I used UTC because there are multiple reports on the web (e.g. [SO, edit from 2017](https://stackoverflow.com/a/6117889/1741604)) that not using UTC may result in incorrect weeks. Additionally I refactored the week calculation a bit based on code from moment.js (which also uses UTC) to really fix non-ISO8601 calendar views (and issue #1052).